src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageReadmeControl.xaml.cs

```diff
@@ -22,16 +25,47 @@ public partial class PackageReadmeControl : UserControl
 #pragma warning disable CS0618 // Type or member is obsolete
         private IMarkdownPreview _markdownPreview;
 #pragma warning restore CS0618 // Type or member is obsolete
-
+        private bool _isInitializing = false;
         public PackageReadmeControl()
         {
             InitializeComponent();
-            _markdownPreview = MarkdownPreviewSingleton.GetInstance();
-            descriptionMarkdownPreview.Content = _markdownPreview.VisualElement;
         }
 
         public ReadmePreviewViewModel ReadmeViewModel { get => (ReadmePreviewViewModel)DataContext; }
 
+        private async Task InitializeAsync()
+        {
+            if (_markdownPreview is null && !_isInitializing)
+            {
+                _isInitializing = true;
+                await TaskScheduler.Default;
+                IEditorOptionsFactoryService editorOptionsFactoryService;
+                try
+                {
+                    var componentModel = await AsyncServiceProvider.GlobalProvider.GetComponentModelAsync();
+                    editorOptionsFactoryService = componentModel?.GetService<IEditorOptionsFactoryService>();
+                }
+                finally
+                {
+                    await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+                }
+                if (editorOptionsFactoryService is not null)
+                {
+#pragma warning disable CS0618 // Type or member is obsolete
+                    var previewBuilder = new PreviewBuilder();
+#pragma warning restore CS0618 // Type or member is obsolete
+                    previewBuilder.EditorOptions = editorOptionsFactoryService.GlobalOptions;
+                    _markdownPreview = previewBuilder.Build();
```

Comment:
Should markdown to built off the UI thread? I believe Amadeusz Wieczorek mentioned this can take a while.

Reply:
I think that would happen when we call `_markdownPreview.UpdateContentAsync`, I can check with him if it's possible to call that off of the UI thread.

Reply:
I just tried it and UpdateContentAsync fails, since it eventually updates a webview/daytona with the results from Markdig I imagine it expects to be on the UI thread.

Reply:
Okay turns out I forgot that I was accessing the VM. I made a change to allow it to run UpdateContentAsync  in the background. UpdateContentAsync doesn't take a cancellation token so unfortunately once I make the request to load the markdown it update the UI when it's complete.